src/components/common/UserDropdown/index.tsx

```diff
@@ -236,7 +236,9 @@ export const UserDropdown: React.FC<Props> = (props) => {
           <DisconnectButton
             buttonType={ButtonType.danger}
             onClick={() => {
-              disconnect()
+              const c: any = connector
+              if (typeof c.close === 'function') c.close()
```

Comment:
Do you understand in which cases the connector does not have this function close?
It seems uniswap always relies on it:
https://github.com/Uniswap/uniswap-interface/blob/09f30ce0f7043399469644ba80763d636ae5a503/src/components/AccountDetails/index.tsx#L309

Reply:
@josojo 
It depends on the provider, injected provided doesn't have that method. The type is `AbstractConnector`, this is extended and for WalletConnectProvider is available. `AbstractConnector` always define a `deactivate` method. 

In uniswap, for injected/metamask I don't get the disconnect button.

Reply:
I like it better to have also a disconnect button for injected providers... 
I don't understand why one would not want to have it. Do you?